packages/cdk/lib/riff-raff-yaml.ts

```diff
@@ -0,0 +1,236 @@
+	if (!s3Key) {
+		throw new Error('Only S3 based lambdas are supported');
+	}
+	return s3Key.split('/').at(-1) ?? inferredFilename;
+};
+
+export class RiffRaffYamlFile {
+	private readonly riffRaffYaml: RiffRaffYaml;
+	private readonly outDir: string;
+	private readonly guStacks: GuStack[];
+
+	private getRegions(): Set<string> {
+		return new Set(
+			this.guStacks.map(({ region }) => {
+				if (Token.isUnresolved(region)) {
+					throw new Error('Region is a token');
+				}
+				return region;
+			}),
+		);
+	}
+
+	private getStages(): Set<string> {
+		return new Set(this.guStacks.map((_) => _.stage));
+	}
+
+	private getLambdas(stack: GuStack): GuLambdaFunction[] {
+		return stack.node
+			.findAll()
+			.filter(
+				(construct) => construct instanceof GuLambdaFunction,
+			) as GuLambdaFunction[];
+	}
+
+	private getUploadLambdaDeployment(
+		lambda: GuLambdaFunction,
+	): RiffRaffDeployment {
+		const { app } = lambda;
+
+		const fileName = getLambdaFileName(lambda);
+		const { stack } = lambda.stack as GuStack;
+
+		return {
+			name: `${app}-upload-${stack}`,
+			props: {
+				type: 'aws-lambda',
+				stacks: new Set([stack]),
+				app,
+				contentDirectory: path.parse(fileName).name,
+				parameters: {
+					bucketSsmLookup: true,
+					lookupByTags: true,
+					fileName,
+				},
+				actions: ['uploadLambda'],
+			},
+		};
+	}
+
+	private getUpdateLambdaDeployment(
+		lambda: GuLambdaFunction,
+	): RiffRaffDeployment {
+		const { app } = lambda;
+
+		const fileName = getLambdaFileName(lambda);
+		const { stack } = lambda.stack as GuStack;
+
+		return {
+			name: `${app}-update-${stack}`,
+			props: {
+				type: 'aws-lambda',
+				stacks: new Set([stack]),
+				app,
+				contentDirectory: path.parse(fileName).name,
+				parameters: {
+					bucketSsmLookup: true,
+					lookupByTags: true,
+					fileName,
+				},
+				actions: ['updateLambda'],
+			},
+		};
+	}
+
+	private getCloudFormationDeployment({
+		stack,
+		templateFile,
+		constructor: { name },
+	}: GuStack): RiffRaffDeployment {
+		const kebabClassName = name
+			.replace(/([a-z])([A-Z])/g, '$1-$2')
+			.toLowerCase();
+		const deploymentName = [kebabClassName, 'cfn', stack].join('-');
+
+		return {
+			name: deploymentName,
+			props: {
+				type: 'cloud-formation',
+				stacks: new Set([stack]),
+				app: kebabClassName,
+				contentDirectory: `${this.outDir}`,
+				parameters: {
+					templatePath: templateFile,
+				},
+			},
+		};
+	}
+
+	constructor(app: App) {
+		const { outdir, node } = app;
+		this.outDir = outdir;
+
+		const stacks: GuStack[] = node
+			.findAll()
+			.filter((construct) => construct instanceof GuStack) as GuStack[];
+
+		this.guStacks = stacks;
+
+		const regions = this.getRegions();
+		const allowedStages = this.getStages();
+
+		const deployments = new Map<
+			RiffRaffDeploymentName,
+			RiffRaffDeploymentProps
+		>();
+
+		stacks.forEach((stack) => {
+			const lambdas = this.getLambdas(stack);
```

Comment:
Should we add this to GuCDK, we'd also want to find ASGs to create an `autoscaling` Riff-Raff deployment.